### ISSUE TITLE
MacOs Fusion BigSur - Vagrant encountered an unexpected communications error with the Vagrant VMware Utility driver

### ISSUE THREAD

**Issue by ChrisTheTrainer** [hashicorp/vagrant#12195]
Hey there,
I installed Vagrant 2.2.14 + vmware_desktop plugin + licence + vagrant-vmware-utility on my Macbook with BigSur and VMware Fusion 12 Pro. (didn’t use brew, all via dowload pkg and install manually)

I verified that the vagrant-vmware-utility is running via:

% sudo launchctl list |grep -i vagrant

0 com.vagrant.vagrant-vmware-utility-stopper
2 com.vagrant.vagrant-vmware-utility
I also restarted it via “unload” and “load”.

But still - if I try starting anything via “vagrant up” I get:
% vagrant up

Vagrant encountered an unexpected communications error with the
Vagrant VMware Utility driver. Please try to run the command
again. If this error persists, please contact support@hashicorp.com

I googled a lot - I tried to “sudo vagrant up” but no luck.

I also don’t see any vagrant processes while the vagrant vmware utility service is running - isn’t that odd as well?

Any tip for me?

**Comment by ChrisTheTrainer** [comment]
I tried A LOT - it's "FIXED" (or worked around) now by simply downgrading from 1.0.18 to 1.0.17 of the vagrant-vmware-utility.

I HOPE you guys will release a fixed 1.0.19 soon

EDIT: I switched to virtualbox provider finally as another issue came right up with fusion after this one and fusion ui is not properly showing vagrant vms :(

**Comment by jnardiello** [comment]
+1, I'm having exactly the same problem. Downgrading the plugin will have vagrant to boot VMs but then another issue kicks-in (networking related, vagrant won't connect to the booted up VM).

So, any hint on how to solve this?

**Comment by jnardiello** [comment]
After a bit of digging, I guess something very wrong happened with the release process. Trying to load manually the `vmware-utility-service`, after failing to load for some wrong permissions on `settings/nat.json` and `settings/portforwarding.json`, fails with this stacktrace:

```
 !  ~  /opt/vagrant-vmware-desktop/bin/vagrant-vmware-utility api -port=9922 -config-file=/opt/vagrant-vmware-desktop/config/service.hcl
2021-02-22T02:04:44.044+0100 [INFO]  vagrant-vmware-utility.api: created: vmx=<nil>
2021-02-22T02:04:44.118+0100 [INFO]  vagrant-vmware-utility.api: attempting to upgrade to vmrest driver
2021-02-22T02:04:44.118+0100 [WARN]  vagrant-vmware-utility.api.vmrest: standard vmware license detected, using fallback
2021-02-22T02:04:44.118+0100 [INFO]  vagrant-vmware-utility.api: enabling internal port forwarding service
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xd8 pc=0x1508ef6]

goroutine 1 [running]:
main.cleanPanic()
        /root/481086944-81/go_src/vagrant-vmware-utility/main.go:66 +0x89
panic(0x15a2b20, 0x1afec30)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/hashicorp/vagrant-vmware-desktop/go_src/vagrant-vmware-utility/driver.(*VmrestDriver).EnableInternalPortForwarding(0x0, 0x164fa06, 0x29)
        /root/481086944-81/go_src/vagrant-vmware-utility/driver/vmrest.go:475 +0x26
github.com/hashicorp/vagrant-vmware-desktop/go_src/vagrant-vmware-utility/command.(*RestApiCommand).buildRestApi(0xc000196310, 0x0, 0x0, 0x26c2, 0x0, 0x0, 0xc000180300)
        /root/481086944-81/go_src/vagrant-vmware-utility/command/rest_api.go:156 +0x8b8
github.com/hashicorp/vagrant-vmware-desktop/go_src/vagrant-vmware-utility/command.(*RestApiCommand).Run(0xc000196310, 0xc00001e0e0, 0x2, 0x2, 0xc000180060)
        /root/481086944-81/go_src/vagrant-vmware-utility/command/rest_api.go:71 +0x8f
github.com/mitchellh/cli.(*CLI).Run(0xc000186000, 0xc000186000, 0x1754020, 0xc000182000)
        /root/go/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x1da
main.realMain(0xc000000008)
        /root/481086944-81/go_src/vagrant-vmware-utility/main.go:53 +0x21a
main.main()
        /root/481086944-81/go_src/vagrant-vmware-utility/main.go:28 +0x105
```

Hope it helps.

**Comment by ChrisTheTrainer** [comment]
Same here for me - very unsatisfying after paying money for the fusion plugin only to going back to vbox for now

**Comment by chrisroberts** [comment]
Hi there. I'm trying to get a proper reproduction of this locally. Would you let me know what version of Big Sur you are currently running on. Thanks!

**Comment by jnardiello** [comment]
Hi @chrisroberts - My version is: `11.2.1 (20D74)` (latest?)

**Comment by chrisroberts** [comment]
@ChrisTheTrainer Can you provide the output found in the `/Library/Application Support/vagrant-vmware-utility/service.log` when using the 1.0.18 version and it's not loading as expected. I've been trying to reproduce the behavior locally starting with Big Sur 11.0.1 and on 11.2.1 and have so far been unable to. This is with a fresh install as well as starting with a 1.0.17 install of the vagrant-vmware-utility and upgrading to 1.0.18. The contents in the log file should hopefully provide a clue as to the origin of the issue.

@jnardiello I've also tried to reproduce the error you encountered and have been unable to as well. You note that the permissions on various files were incorrect, but I do not believe that is the case. The vagrant-vmware-utility process runs as a privileged user so those files should not have required a permissions modification. It appears you are attempting to run the utility manually as a user but if you are running it manually you'll need to either be root, or use `sudo`

**Comment by chrisroberts** [comment]
@ChrisTheTrainer Actually, I was finally able to track down a modification that would force the panic provided by @jnardiello and so now have a valid reproduction. If either of you could provide the output from running:

```
sudo /opt/vagrant-vmware-desktop/bin/vagrant-vmware-utility api -debug
```

that would be very helpful. The driver downgrade during startup is triggering the error, which I'll resolve. But if you are using a pro vmware license, vmrest should be available and the downgrade shouldn't be necessary. If the license content has changed somehow, I'd like to make sure that is addressed as well before releasing a fix.

Thanks!

**Comment by ChrisTheTrainer** [comment]
Is this expected:

% sudo /opt/vagrant-vmware-desktop/bin/vagrant-vmware-utility api -debug
Password:
2021-02-24T11:02:24.369+0100 [INFO]  vagrant-vmware-utility.api: created: vmx=<nil>
2021-02-24T11:02:24.371+0100 [TRACE] vagrant-vmware-utility.api.settings: building nat settings: path=/opt/vagrant-vmware-desktop/settings/nat.json
2021-02-24T11:02:24.372+0100 [DEBUG] vagrant-vmware-utility.api: loading vmware information
2021-02-24T11:02:24.372+0100 [TRACE] vagrant-vmware-utility.api: vmware version check: vmx-path="/Applications/VMware Fusion.app/Contents/Library/vmware-vmx"
2021-02-24T11:02:24.843+0100 [DEBUG] vagrant-vmware-utility.api: initial vmware information loaded: license=fusion.ws.pro.vl
2021-02-24T11:02:24.843+0100 [DEBUG] vagrant-vmware-utility.api: normalized vmware information: license=standard
2021-02-24T11:02:24.843+0100 [TRACE] vagrant-vmware-utility.api: returning cached vmware information
2021-02-24T11:02:24.843+0100 [TRACE] vagrant-vmware-utility.api: returning cached vmware information
2021-02-24T11:02:24.843+0100 [DEBUG] vagrant-vmware-utility.api: creating new advanced driver
2021-02-24T11:02:24.843+0100 [INFO]  vagrant-vmware-utility.api: attempting to load vmrest driver
2021-02-24T11:02:24.843+0100 [TRACE] vagrant-vmware-utility.api: returning cached vmware information
2021-02-24T11:02:24.843+0100 [WARN]  vagrant-vmware-utility.api.vmrest: standard vmware license detected, using fallback
2021-02-24T11:02:24.843+0100 [TRACE] vagrant-vmware-utility.api: running VMware Fusion app bundle validation
Starting the Vagrant VMware API service
2021-02-24T11:02:25.892+0100 [DEBUG] vagrant-vmware-utility.api.api: start api service requested
2021-02-24T11:02:25.892+0100 [TRACE] vagrant-vmware-utility.api.api: registering routes
2021-02-24T11:02:25.892+0100 [INFO]  vagrant-vmware-utility.api.api: api service start: host=127.0.0.1 port=9922
Failed to start the Vagrant VMware API service - listen tcp 127.0.0.1:9922: bind: address already in use
cb@F8-FF-C2-36-1D-51 ~ %

**Comment by ChrisTheTrainer** [comment]
(NOTE: I had to uninstall the service first before doing this as it was already listening on 9922)

**Comment by chrisroberts** [comment]
@ChrisTheTrainer Perfect, thank you! And sorry about not being clear with running the command. It was okay that it didn't stay running. I just needed to see the debug output during setup to confirm where I thought the issue was occurring. I should have this resolved and a release pushed out before the end of the day today.

**Comment by jnardiello** [comment]
@chrisroberts Is there any way we can still help?

**Comment by chrisroberts** [comment]
@jnardiello Hi! I'm just wrapping up testing on the various platforms. I got held up last night waiting on a catalina update to finish. Once I've finished this up this morning everything will be ready to go.

**Comment by jnardiello** [comment]
No worries! Thank you for your work and let us know if there's any other way we can help.

**Comment by chrisroberts** [comment]
A new utility release has been published that addresses this issue. You can get it here:

https://releases.hashicorp.com/vagrant-vmware-utility/1.0.19/

If you have any issues, please do let me know. I'm going to close this issue, but I'll keep an eye on it for a bit if there are any problems. Cheers!

**Comment by coxoperationsengineer** [comment]
@chrisroberts I can confirm the same setup, was having the same issue, upgraded to 1.0.19, and all is well in the universe of VMWare Fusion Pro + Vagrant + VMWare provider + Big Sur  :)   Thank you for jumping on this!

**Comment by ChrisTheTrainer** [comment]
Didn’t know I could still write here as it was closed, so same here, works great thank you a lot

**Comment by jnardiello** [comment]
Works great, thanks @chrisroberts for the fix and @mitchellh for pushing this internally <3
